api-consumption/src/main/java/org/zalando/nakadi/service/subscription/autocommit/PartitionState.java

```diff
@@ -0,0 +1,67 @@
+package org.zalando.nakadi.service.subscription.autocommit;
+
+import org.zalando.nakadi.domain.NakadiCursor;
+import org.zalando.nakadi.service.CursorOperationsService;
+
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * There are several assumptions on behavior of this class.
+ * 1. Offsets to skip are only added to the end of skipped events queue.
+ * 2. Commits are added at any place, and on every commit we are removing queue of intervals to skip that are
+ * completely before committed offset.
+ * Thus, if we are about to do autocommit, we should check first interval only.
+ * 3. Committed offsets are only increasing (this is guaranteed by other part of the code)
+ */
+public class PartitionState {
+    private final CursorOperationsService cursorOperationsService;
```

Comment:
PartitionState looks like a model at least from the name perspective, I would expect to have additional class which is able to work with it instead of having cursorOperationsService inside the data class.

Reply:
I already shared my opinion that essential part of OOP approach requires joining data (or object state) with methods of it's processing. 
In order to make it separate there should be some reasons. Here I don't really see this reasons.

But I agree that class name is somehow misleading. How do you think `PartitionSkippedCursorsOperator` name reflects properly the actions that are being executed by this class?

Reply:
@adyach  ?